src/NServiceBus.Metrics/QueueLength/QueueLengthMarticBuilder.cs

```diff
@@ -0,0 +1,157 @@
+﻿namespace NServiceBus.Metrics.QueueLength
+{
+    using System;
+    using System.Collections.Concurrent;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Features;
+    using global::Metrics;
+    using global::Metrics.Core;
+    using Pipeline;
+    using Routing;
+
+    class QueueLengthMetricBuilder : MetricBuilder
+    {
+        const string HeaderName = "NServiceBus.Metrics.QueueLength";
+
+        const char SeparatorValue = '_';
+        static readonly string Separator = SeparatorValue.ToString();
+        static readonly char[] Separators = { SeparatorValue };
+
+        MetricsContext metricsContext;
+
+        ConcurrentDictionary<string, Tuple<Guid, CounterImplementation>> sendingCounters = new ConcurrentDictionary<string, Tuple<Guid, CounterImplementation>>();
+        ConcurrentDictionary<Guid, SequenceReporter> receivingReporters = new ConcurrentDictionary<Guid, SequenceReporter>();
+        static readonly Unit Unit = Unit.Custom("Sequence");
+
+        public override void Define(MetricsContext metricsContext)
```

Comment:
By not using the attributes based approach the metadata is not readable. So that means the counter would not be exposed in the performance counter pkg. Is that something you plan to address in a different PR?